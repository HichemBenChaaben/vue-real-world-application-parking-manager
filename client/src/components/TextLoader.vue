<template>
  <div class="h-4 bg-gray-200 rounded w-24 animate-pulse"></div>
</template>
<script lang="ts" setup></script>
