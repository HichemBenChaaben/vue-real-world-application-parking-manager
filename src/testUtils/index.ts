import { VueWrapper, mount, type MountingOptions } from '@vue/test-utils'
import type { Component, ComponentPublicInstance } from 'vue'

/**
 * a wrapper for the mount function that magically knows about component props
 * via wrapper.vm.$props[]
 * example wrapper.vm.$props['your-props'] will get you access to the value it self
 * extends vue-test-utils mount function
 */
export const factory = <P, D = Record<string, symbol>>(
  Component: Component,
  options?: MountingOptions<P, D>
): VueWrapper<ComponentPublicInstance<P>> => {
  return mount(Component, {
    ...options
  })
}

/**
 * function to return element by tesId
 * usage
 * const  { findByTestId } = fromWrapper(wrappper);
 * const el = findByTestId('your-test-id');
 *
 * @param wrapper VueWrapper
 */
export const fromWrapper = (wrapper: VueWrapper) => {
  const findByTestId = (testid: string) => wrapper.find(`[data-testid="${testid}"]`)
  return { findByTestId }
}
