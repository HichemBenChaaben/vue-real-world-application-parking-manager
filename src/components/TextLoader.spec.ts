import { describe, test, expect, beforeEach } from 'vitest'
import type { Component, ComponentPublicInstance } from 'vue'
import type { VueWrapper } from '@vue/test-utils'
import TextLoader from './TextLoader.vue'
import { factory } from '@/testUtils'

/**
 * most of type generics do not need to explicetely be defined like this
 * i just did it to show how it works if typescript wouldnt auto-infer the type
 */

const mountFunction = <T extends Component>(component: T) => factory(component)

describe('TextLoader', () => {
  let wrapper: VueWrapper<ComponentPublicInstance>

  beforeEach(() => {
    wrapper = mountFunction<Component>(TextLoader)
  })

  test('render and match snapshot', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('contains animate-pulse class', () => {
    expect(wrapper.classes()).toContain('animate-pulse')
  })
})
