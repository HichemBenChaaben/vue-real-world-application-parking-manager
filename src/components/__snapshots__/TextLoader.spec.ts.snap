// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextLoader > render and match snapshot 1`] = `"<div class="h-4 bg-gray-200 rounded w-24 animate-pulse"></div>"`;
